core/services/ocr2/plugins/directrequestocr/reporting.go

```diff
@@ -273,9 +272,16 @@ func (r *functionsReporting) Report(ctx context.Context, ts types.ReportTimestam
 			continue
 		}
 
-		// TODO: support per-request aggregation method
-		// https://app.shortcut.com/chainlinklabs/story/57701/per-request-plugin-config
-		aggregated, errAgg := Aggregate(defaultAggMethod, observations)
+		request, err := r.pluginORM.FindById(sliceToByte32(observations[0].RequestID), pg.WithParentCtx(ctx))
+		if err != nil {
+			r.logger.Debug("FunctionsReporting Observation can't find request from query", commontypes.LogFields{
+				"requestID": reqId,
+				"err":       err,
+			})
+			continue
```

Comment:
This is trickier. If a node doesn't have a request locally but has enough observations from other nodes, it should still aggregate. Otherwise nodes won't agree on the final report (and with batching it lead to a deadlock).

It looks like we need to pass agg method as part of an observation (again maybe wrap in a metadata sub-struct?). Then, when aggregating we should first pick the agg method using MODE (loud error if they aren't all equal) and then aggregate using that method.